From 799f75cdeafc0ba124b7ebc89f21a873a015e2e7 Mon Sep 17 00:00:00 2001
From: tedyu <yuzhihong@gmail.com>
Date: Mon, 14 Mar 2016 07:09:58 -0700
Subject: [PATCH 142/308] HBASE-15439 getMaximumAllowedTimeBetweenRuns in
 ScheduledChore ignores the TimeUnit (Jingcheng Du)

Change-Id: I1bf1e2d1e934b57889e222842077eeeaa43bbfbc
Reason: Bug
Author: Jingcheng Du
Ref: CDH-38410
---
 .../org/apache/hadoop/hbase/ScheduledChore.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java b/hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java
index dd98d26..2c54b9f 100644
--- a/hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java
+++ b/hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java
@@ -230,7 +230,7 @@ public abstract class ScheduledChore implements Runnable {
 
   private double getMaximumAllowedTimeBetweenRuns() {
     // Threshold used to determine if the Chore's current run started too late
-    return 1.5 * period;
+    return 1.5 * timeUnit.toMillis(period);
   }
 
   private synchronized boolean isValidTime(final long time) {
-- 
1.7.9.5

